connect/runtime/src/test/java/org/apache/kafka/connect/integration/ConnectorValidationIntegrationTest.java

```diff
@@ -0,0 +1,493 @@
+    public void testSinkConnectorHasBothTopicsListAndTopicsRegex() {
+        Map<String, String> config = defaultSinkConnectorProps();
+        config.put(TOPICS_CONFIG, "t1");
+        config.put(TOPICS_REGEX_CONFIG, "r.*");
+        connect.validateConnectorConfig(config.get(CONNECTOR_CLASS_CONFIG), config);
+    }
+
+    @Test
+    public void testSinkConnectorDeadLetterQueueTopicInTopicsList() {
+        Map<String, String> config = defaultSinkConnectorProps();
+        config.put(TOPICS_CONFIG, "t1");
+        config.put(DLQ_TOPIC_NAME_CONFIG, "t1");
+        connect.validateConnectorConfig(config.get(CONNECTOR_CLASS_CONFIG), config);
+    }
+
+    @Test
+    public void testSinkConnectorDeadLetterQueueTopicMatchesTopicsRegex() {
+        Map<String, String> config = defaultSinkConnectorProps();
+        config.put(TOPICS_REGEX_CONFIG, "r.*");
+        config.put(DLQ_TOPIC_NAME_CONFIG, "ruh.roh");
+        connect.validateConnectorConfig(config.get(CONNECTOR_CLASS_CONFIG), config);
+    }
+
+    @Test
+    public void testSinkConnectorDefaultGroupIdConflictsWithWorkerGroupId() throws InterruptedException {
+        Map<String, String> config = defaultSinkConnectorProps();
+        // Combined with the logic in SinkUtils::consumerGroupId, this should conflict with the worker group ID
+        config.put(NAME_CONFIG, "worker-group-id");
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(
+            config.get(CONNECTOR_CLASS_CONFIG),
+            config,
+            1,
+            "Sink connector config should fail preflight validation when default consumer group ID conflicts with Connect worker group ID"
+        );
+    }
+
+    @Test
+    public void testSinkConnectorOverriddenGroupIdConflictsWithWorkerGroupId() throws InterruptedException {
+        Map<String, String> config = defaultSinkConnectorProps();
+        config.put(CONNECTOR_CLIENT_CONSUMER_OVERRIDES_PREFIX + GROUP_ID_CONFIG, WORKER_GROUP_ID);
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(
+            config.get(CONNECTOR_CLASS_CONFIG),
+            config,
+            1,
+            "Sink connector config should fail preflight validation when overridden consumer group ID conflicts with Connect worker group ID"
+        );
+    }
+
+    @Test
+    public void testSourceConnectorHasDuplicateTopicCreationGroups() throws InterruptedException {
+        Map<String, String> config = defaultSourceConnectorProps();
+        config.put(TOPIC_CREATION_GROUPS_CONFIG, "g1, g2, g1");
+        connect.assertions().assertExactlyNumErrorsOnConnectorConfigValidation(
+            config.get(CONNECTOR_CLASS_CONFIG),
+            config,
+            1,
+            "Source connector config should fail preflight validation when the same topic creation group is specified multiple times"
+        );
+    }
+
+    @Test
+    // TODO: Is this actually necessary? Should we permit the same SMT to be applied multiple times?
```

Comment:
Was this a validation that already takes place, and you're just adding a test for it?
I think this makes sense because you can just work around it by renaming your transform name and copy-pasting the transform configurations.

Reply:
> Was this a validation that already takes place, and you're just adding a test for it?

Yep, exactly.

> I think this makes sense because you can just work around it by renaming your transform name and copy-pasting the transform configurations.

I hear that, but at the same time I wonder if it's doing anybody any good to force them to copy+paste in the first place.